robot/impl/resource_manager.go

```diff
@@ -836,6 +836,15 @@ func (manager *resourceManager) processRemote(
 	config config.Remote,
 	gNode *resource.GraphNode,
 ) (*client.RobotClient, error) {
+	// if there was an existing client (i.e. remote was modified), close old client before making a new one
```

Comment:
can we be sure that `updateRemoteResourceNames` is getting called and correctly replacing the clients?

Reply:
yes, as is. `processRemote` is only called in a code path that is guaranteed to call `updateRemoteResourceNames`

Reply:
the other changes addressed this comment